Character-based gear loadouts

giacomoh opened the pull request:
## About The Pull Request
Adds character-based gear loadouts, instead of the client-based loadouts.
To test if this works, you can edit the cost of clothes in loadout/loadout_(whatever clothes you want to test).
This is my first PR. It worked fine on my end, but there might (surely) be bugs I have overlooked.

## Why It's Good For The Game
Changing your loadouts manually each time you wanted to play another character is a pain, this is less painful.

## Testing Photographs and Procedure

<details>

here's how it looks

<img width="551" alt="cloadout1" src="https://user-images.githubusercontent.com/53836108/167031980-7b889db1-944f-4c7b-b7a9-ed4305cfa029.png">

<img width="551" alt="cloadout2" src="https://user-images.githubusercontent.com/53836108/167031660-389b7173-f514-4f4b-b4d5-dda318852623.png">
72057f1e547.png">

</details>

## Changelog
:cl:
add: Added character selection slots in beecoin shop
add: Added equipped_loadout to save_character
/:cl:

giacomoh committed:
Revert "Character-based gear loadouts"

This reverts commit edb2a7efd466aada3f62ab22512ce36f92aa603d.

giacomoh committed:
Character-based gear loadouts

Adds character selection in Beecoin shop, and saves the equipped loadouts to characters instead of saving it to the client preferences.
I also lowered the cost of two pairs of shoes for testing purposes.
It worked fine on my end, but this is my first PR so there might be some bugs i didn't notice.

giacomoh committed:
Merge remote-tracking branch 'origin/master'

giacomoh committed:
Removed testing negative beecoins

Liidero commented:
Good stuff! Would love go to see this merged!

PowerfulBacon commented on code/modules/client/preferences.dm:
```diff
@@ -724,6 +724,23 @@ GLOBAL_LIST_EMPTY(preferences_datums)
+					dat += "<center>"
+					var/name
+					var/unspaced_slots = 0
+					for(var/i=1, i<=max_save_slots, i++)
```
Use for(var/i in 1 to max_save_slots)

giacomoh replied:
Are you asking me to write "for(var/i in 1 to max_save_slots)" instead of "for(var/i=1, i<=max_save_slots, i++)" ?

EDIT: sorry, i'm not too familiar with what the code you wrote means that's why i'm asking

PowerfulBacon replied:
yea

giacomoh replied:
done 👍

PowerfulBacon reviewed (commented):
Looked at this for a while and I can't figure out what exactly this is doing or why it works

giacomoh committed:
edited as Bacon requested

PowerfulBacon commented:
Could you explain how the code works? I can't quite wrap my head around why this is working

giacomoh commented:
> Could you explain how the code works? I can't quite wrap my head around why this is working

I'll preface my explaination by saying that this was my first time coding anything, so there are probably many mistakes and i don't know the technical terms (at all).

What i first did was making the characters appear as boxes on top. i used similar code that is used for character selection. 
Then I changed where the choices are saved (from preferences to character)
Then i also changed the READ and WRITE _files accordingly.

From testing i didn't have any issues or bugs

PowerfulBacon reviewed (approved):
I see, this makes sense now.

giacomoh commented:
deleting this branch since i'm deleting this fork. I will make a new PR, with all the changes on another fork. Sorry for the inconvenience.
